src/content/docs/cheerp/guides/cli.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Arguments and environment variables
+---
+
+Cheerp (nightly) has support for passing arguments and environment variables to your programs. The way you pass them
+depends on whether you're compiling for WASI, ES6 or something else
+
+For all the examples we'll use the following example program:
+
+```c
+// test.c
+#include <stdio.h>
+
+int main(int argc, char **argv, char **env) {
+  for (int i = 0; i < argc; ++i)
+    printf("arg: %s\n", argv[i]);
+
+  for (int i = 0; env[i]; ++i)
+    printf("env: %s\n", env[i]);
+}
+```
+
+## WASI
+
+Cheerp uses the WASI-CLI interface to get the arguments and environment variables, meaning that it will work on any
+runtime that has implemented the [WASI-CLI proposal](https://github.com/WebAssembly/wasi-cli).
+
+Here we'll be using `wasmtime`:
+
+```
+$ /opt/cheerp/bin/clang -target cheerp-wasm-wasi test.c
+$ wasmtime --env A=1 --env B=2 a.out a b c
+arg: a.out
+arg: a
+arg: b
+arg: c
+env: A=1
+env: B=2
+```
+
+## ES6
+
+For ES6 modules, you can pass the arguments through the initialization function like this:
+
+```js
+// main.mjs
+import init from "./test.mjs";
+
+init({ argv: ["a", "b", "c"], env: ["A=1", "B=2"] }).then((module) => {});
+```
+
+```
+$ /opt/bin/clang test.c -cheerp-make-module-es6 -o test.mjs
+$ node main.mjs
+arg: a
+arg: b
+arg: c
+env: A=1
+env: B=2
+```
+
+## Other
+
+For anything else, you can pass environment variables and program arguments by defining these globals somewhere before
+the call to `main`:
+
+- `CHEERP_ENV`
+- `CHEERP_ARGV`
+
+Do note that changes to `CHEERP_ENV` after `main` has run will not be reflected in `environ`.
```

Comment:
This line seems to imply that changes to argv will be - is this the case? (I expect not, in which case this line should talk about both!)

Reply:
You can't get the arguments from a different place like environment variables (`environ`), only through main, which will only be called once by the start function